scripts/trim-low-abund.py

```diff
@@ -0,0 +1,295 @@
+#! /usr/bin/env python2
+#
+# This file is part of khmer, http://github.com/ged-lab/khmer/, and is
+# Copyright (C) Michigan State University, 2009-2014. It is licensed under
+# the three-clause BSD license; see doc/LICENSE.txt.
+# Contact: khmer-project@idyll.org
+#
+"""
+Trim sequences at k-mers of the given abundance, using a streaming algorithm.
+Output sequences will be placed in 'infile.abundtrim'.
+
+% python scripts/trim-low-abund.py [ <data1> [ <data2> [ ... ] ] ]
+
+Use -h for parameter help.
+
+TODO: load/save counting table.
+TODO: reference appropriate preprint.
+"""
+import sys
+import screed
+import os
+import khmer
+import tempfile
+import shutil
+import textwrap
+
+from screed.screedRecord import _screed_record_dict
+from khmer.khmer_args import build_counting_args
+from khmer.utils import (write_record, write_record_pair, broken_paired_reader)
+
+
+DEFAULT_NORMALIZE_LIMIT = 20
+DEFAULT_CUTOFF = 2
+
+# see Zhang et al., http://arxiv.org/abs/1309.2975
+MAX_FALSE_POSITIVE_RATE = 0.8
+
+
+def trim_record(read, trim_at):
+    new_read = _screed_record_dict()
+    new_read.name = read.name
+    new_read.sequence = read.sequence[:trim_at]
+    if hasattr(read, 'accuracy'):
+        new_read.accuracy = read.accuracy[:trim_at]
+
+    return new_read
+
+
+def get_parser():
+    epilog = """
+    The output is one file for each input file, <input file>.abundtrim, placed
+    in the current directory.  This output contains the input sequences
+    trimmed at low-abundance k-mers.
+
+    The ``-V/--variable-coverage`` parameter will, if specified,
+    prevent elimination of low-abundance reads by only trimming
+    low-abundance k-mers from high-abundance reads; use this for
+    non-genomic data sets that may have variable coverage.
+
+    Note that the output reads will not necessarily be in the same order
+    as the reads in the input files; if this is an important consideration,
+    use ``load-into-counting.py`` and ``filter-abund.py``.  However, read
+    pairs will be kept together, in "broken-paired" format; you can use
+    ``extract-paired-reads.py`` to extract read pairs and orphans.
+
+    Example::
+
+        trim-low-abund.py -x 5e7 -k 20 -C 2 data/100k-filtered.fa
+    """
+
+    parser = build_counting_args(
+        descr='Trim low-abundance k-mers using a streaming algorithm.',
+        epilog=textwrap.dedent(epilog))
+
+    parser.add_argument('--cutoff', '-C', type=int, dest='cutoff',
```

Comment:
*sigh* The point is that you can leave the 'dest' off and it'll match the long form of the argument :-)